First batch of improvements.

Posiplusive opened the pull request:
This PR adds:
- MKV file tagging which adds group name and description to the MKV video.
New dependency: mkvpropedit
- `description` property for Episode() and Movie() classes.
- Bugfix where caching to db prints 1 key despite having cached multiple keys.
- (Minor) ignore services directory in git.
- (Minor) README updates to refer to this fork in particular instead of devine.

Planned features:
- mp4decrypt support
- N_m3u8DL-RE downloader support for faster downloads.

Shivelight commented on vindemitor/core/post_processor.py:
```diff
@@ -101,15 +103,10 @@ def _attach_fonts(self, title: Title_T):
             self.log.info(f"Attached {font_count} fonts for the Subtitles")
 
     def _repackage_tracks(self, title: Title_T):
-        has_repacked = False
         for track in title.tracks:
             if track.needs_repack:
                 track.repackage()
-                has_repacked = True
                 events.emit(events.Types.TRACK_REPACKED, track=track)
-        if has_repacked:
-            # we don't want to fill up the log with "Repacked x track"
-            self.log.info("Repacked one or more tracks with FFMPEG")
```
Is this necessary?

Posiplusive replied:
Not really, but would be nice to see all the tracks' statuses. I can revert it if you liked the old one.

Shivelight replied:
Isn't this just silencing the log? I don't see the statuses are logged anywhere else.

Shivelight commented on vindemitor/core/post_processor.py:
```diff
@@ -128,3 +125,19 @@ def _mux_media(self, title: Title_T, tracks: Tracks, progress_callback) -> Path:
+
+    def _tag_file(self, path, tag, title: Title_T):
+        xml_lines = ['<?xml version="1.0" encoding="UTF-8"?>', "<Tags>", "  <Tag>", "    <Targets/>"]
+        xml_lines.append(f"    <Simple><Name>Description</Name><String>{title.description}</String></Simple>")
```
This will raise `AttributeError` when `Song` is passed. Either guard it or add `description` attribute to `Song` class, whichever made more sense.

Shivelight commented on vindemitor/core/post_processor.py:
```diff
@@ -128,3 +125,19 @@ def _mux_media(self, title: Title_T, tracks: Tracks, progress_callback) -> Path:
         return muxed_path
+
+    def _tag_file(self, path, tag, title: Title_T):
+        xml_lines = ['<?xml version="1.0" encoding="UTF-8"?>', "<Tags>", "  <Tag>", "    <Targets/>"]
+        xml_lines.append(f"    <Simple><Name>Description</Name><String>{title.description}</String></Simple>")
+        xml_lines.append(f"    <Simple><Name>Group</Name><String>{tag}</String></Simple>")
+        xml_lines.extend(["  </Tag>", "</Tags>"])
```
Use multiline f-string literal

Shivelight commented on vindemitor/core/post_processor.py:
```diff
@@ -128,3 +125,19 @@ def _mux_media(self, title: Title_T, tracks: Tracks, progress_callback) -> Path:
+            f.write("\n".join(xml_lines))
+            tmp_path = Path(f.name)
+        subprocess.run(
+            ['mkvpropedit', path, '--tags', f'global:{tmp_path}'],
```
See `vindemitor\core\binaries.py`.
This code should not crash for any reason (e.g. missing binary). Execution must continue.

Posiplusive commented:
Okay, I improved some stuffs related to mkvpropedit and added proper checks now. The `_tag_file()` post-processor won't run if it's a Song(), or if it's not a Song() but there's no mkvpropedit, execution will continue as normal. In addition to that, I also added a check for mkvmerge in `tracks.mux()` and will raise EnvironmentError if it's not found.

<img width="626" height="149" alt="image" src="https://github.com/user-attachments/assets/22615334-3527-4002-a448-7e52c5bf0a9e" />
<img width="724" height="139" alt="image" src="https://github.com/user-attachments/assets/99f6037c-24f0-4f14-8924-e4198c5e3e99" />

Posiplusive commented:
Okay, I delegated the binaries checking task to the post-processors, `_mux_media()` and `_tag_file()`. I also used a real multiline f-string this time.

Shivelight commented:
Merged
- e8bff83f755584516fa5928f21637b5bb4fed09c
- 699284cb875ba6eeef8c3a1e1754a00d1e7a4522
- 4f04340c973910104acef54f69c4c24d20a8dc4c
- c1391d8aeb97cb693182df961564ff6e2fed1c9d
